misk/src/main/kotlin/misk/moshi/wire/FieldBinding.kt

```diff
@@ -0,0 +1,118 @@
+package misk.moshi.wire
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.wire.Message
+import com.squareup.wire.WireField
+import java.lang.reflect.Field
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+
+internal class FieldBinding(
+        wireField: WireField,
+        builderType: Class<Message.Builder<*, *>>,
+        private val messageField: Field,
+        private val moshi: Moshi
+) {
+    val name: String = messageField.name
+    private val label = wireField.label
+    private val keyAdapterString = wireField.keyAdapter
+    private val builderField = getBuilderField(builderType, name)
+    private val builderMethod = getBuilderMethod(builderType, name, messageField.type)
+    private val isMap = keyAdapterString.isNotEmpty()
+    val isList: Boolean = label == WireField.Label.PACKED || label == WireField.Label.REPEATED
+
+    // NB(mmihic): These need to be lazy because there might be cycles in the type graph,
+    // and we don't want to go into infinite recursion
```

Comment:
I _think_ Moshi will do this for you?

Reply:
It does, assuming you're doing all the adapter lookups inside the factory method.
https://github.com/square/moshi/blob/master/moshi/src/main/java/com/squareup/moshi/Moshi.java#L85

Reply:
Neat! Removed the lazy